### PR TITLE
Make button and roller actions configurable for video mode

### LINKED ISSUE #357: Head Tracker CENTERING function on Func button. Feature request
I am so impressed by the Head Tracker implementation, the wireless solution in particular.
Recently I had the opportunity to set it up and to use it.
Though I am really missing a convenient way to center the tracker instantly, whenever desirable. The only way available today, AFAIK is entering the HT menu and choose enter the CENTER option - and in the settings I operate this is a hassle affecting the overall HT experience.

I suggest there are some factors contributing to a great HT experience:

1. Smooth and precise sensing and output generation (check - works fine)
2. Quick and easy way of HT centering at the push of a button (not there yet)
3. Quick and easy calibration (not there yet). 
Calibration may not be necessary every session, but due to temperature variations it might be.

**Suggested solution**
Allow, via goggles configuration menu, to assign the goggles Func. button to control HT CENTER and CALIBRATE - in stead of FAN control.
On a side note - so far, I never experienced a need to adjust the fan speed, especially not during a session. While the need of HT calibration and centering is urgent.

Todays Func button default is, short press - DVR record/stop, long press - FAN speed.
I suggest fully configurable options of, SHORT PRESS, double short press  - "DOUBLE CLICK", and LONG PRESS.

Personally I would configure DOUBLE CLICK to HT CENTER and LONG PRESS to HT CALIBRATE.
SHORT PRESS would remain record on/off.

Oh, and this need of easy centering and calibration - it´s not just me. Look at the Skyzone 04X HT for instance. Dedicated HT center / calibration button on the goggles, ready to use at any moment on ground or airborne. It´s there for a reason.

### PR DESCRIPTION
**Pull request by Master92**
This adds a page in the settings where one can freely assign the actions that are executed when a button or the roller is operated in video mode.
I also added the ability to change the top fan speed and the OLED brightness level with the roller.

When changing a channel is requested, the left short button acts as the "Enter" button, regardless of which action is configured for it.

**Note:** This changes the behavior introduced with #348 where I double-assigned the double right click for either HT Center or Go Sleep! mode. However, as this PR is not yet in the release candidate, I think, this shouldn't do much harm. Closes #357 

The available actions are for the roller:

- Switch channel
- Change fan speed
- OLED brightness

and for the buttons:

- Toggle OSD
- Main menu
- Toggle DVR
- Center HT
- Calibrate HT
- Go Sleep!
- Toggle fan speed

Setting

Roller

Button

### CODE REVIEW

**Comment by SumolX** [comment]
Very nice!  I was working on a similar approach function pointers and all.  Excellent work!!

**Comment by MrMap** [comment]
The functions of this PR is brilliant and very welcome for several reasons. 
As a Head Track user I greatly appreciate the HT Center and HT Calibrate - well done.

**I just got to think about an additional HT option for configurable button assignment: HT on/off.** 
Today I use INAV logical programming to achieve the same (which also needs one radio channel). 
I think there might be multiple situations where the HT user would benefit from easily toggling HT on/off easily on the goggles.

**Comment by Master92** [comment]
> **I just got to think about an additional HT option for configurable button assignment: HT on/off.** 

Good idea. I think, some feedback in the OSD would be needed as well wouldn't it? Simply toggling would be easy but without any feedback it could be pretty hard to guess what's happening.

**Comment by MrMap** [comment]
> > **I just got to think about an additional HT option for configurable button assignment: HT on/off.**
> 
> Good idea. I think, some feedback in the OSD would be needed as well wouldn't it? Simply toggling would be easy but without any feedback it could be pretty hard to guess what's happening.

Quite right! Feedback is desirable.
This also applies for HT calibration; let it beep while calibration in progress (meaning don't move goggles until calibration complete).

**Comment by ligenxxxx** [comment]
This branch has conflicts that must be resolved.

**Comment by Master92** [comment]
Rebased onto current `main`.

**Comment by evilC** [comment]
I just flashed this today, and Go Sleep does not seem to work.
To be clear, I actually flashed [this PR](https://github.com/hd-zero/hdzero-goggle/pull/368) because I wanted both of these things - so conceivably it may have crept in as part of that.
When in the main menu, Go Sleep seems to do absolutely nothing.
If I configure LP of right button as a shortcut for Go Sleep, and then activate it while on the FPV view (But with no connection currently active - ie quad is off), then the following happens:
1. Screen off for a second or 2
2. Screen on and VERY brief loading bar
3. Screen off for about a second
4. Screen on, back to normal

**Comment by Master92** [comment]
Huh, that seems strange. To be fair, I never used HDMI out, yet. Therefore I don't know what the behavior is or was supposed to be.

By saying you wanted to enter sleep mode from the main menu you mean via the lowermost menu entry, right? Because shortcuts and button actions only work in video mode.

Can you confirm that the behavior is similar with a previous release version? If that's the case, please open a new issue.
This PR didn't actually change any code related to sleep mode, therefore I can only imagine it being an already existing issue. The only time I touched sleep code was in #348 while I extracted it from one place to another.

**Comment by evilC** [comment]
That was a complete red herring...
Of course, when in the menu, if you activate sleep, and you are on HDMI (Which does not disable during Go Sleep), then "Nothing happens" is how it is going to be perceived
I never used Go Sleep before, so I cannot comment on whether it changed.
TLDR - nothing to do I don't think. As soon as I realized that sleep mode does not stop feeding HDMI, all the observed behaviour makes complete sense.

**Comment by pitts-mo** [comment]
Hello @Master92 
Are you able to make the input page display the current settings on fresh power up?

I have been running this from the SD card for about a week and find this a very helpful feature. The only concerning observation I have so far is that on fresh boot the button options within the input menu appear to show default settings. However, testing the functions after a power cycle does show my button preferences set during the last power session were saved correctly. Additionally, if I adjust settings my preferences the states displayed are again preserved within the input menu for the remainder of the power session.

**Comment by Master92** [comment]
Ooff, I thought I already did that. So you're saying, everytime you power up, the settings page displays the defaults instead of the current settings but the button actions act as they were configured in the previous session?

I am currently unable to test that but if you can safely verify that that's the case, please open an issue and tag me so we can easier track what's going on.

Happy new year btw. 🎉😃🚀
